scripts/mkimage

```diff
@@ -262,6 +262,14 @@ EOF
     mcopy $RELEASE_DIR/target/KERNEL.md5 "::/$KERNEL_NAME.md5"
     mcopy $RELEASE_DIR/target/SYSTEM.md5 ::/SYSTEM.md5
 
+    MKIMAGE_TOOL_CMD="$TOOLCHAIN/bin/mkimage -A $TARGET_KERNEL_ARCH -O linux -T script -C none -d"
+    for src in $INSTALL_SRC_DIR/*autoscript.src ; do
+      if [ -e "$src" ] ; then
+        $MKIMAGE_TOOL_CMD "$src" "$LE_TMP/$(basename $src .src)" > /dev/null
```

Comment:
change this to
`mkimage -A $TARGET_KERNEL_ARCH -O linux -T script -C none -d "$src" "$LE_TMP/$(basename $src .src)" > /dev/null`

Reply:
It won't work if you don't have mkimage installed, that's why it can be compiled as a package.

Any objections for adding `u-boot-tools:host` as dependency for toolchain?

Reply:
https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/tools/u-boot/package.mk

Same?

Reply:
kszaq, it will I have already tried it.

As long as ...
`
EXTRA_DEPS="mkimage"
EXTRA_DEPS_PKG="u-boot-tools"
`
Is in project/xxx/options the buildscript won't continue unless the host-os has them installed, if they are not installed the user is prompted if they would like to install them, adding u-boot-tools:host isn't needed

Reply:
It all depends how much control you want over the tools you will be using to create your image.

If you're happy to use whatever version of `u-boot-tools` is installed on the host (which may be an an old version, with unknown bugs) then sure, use the host installed version, and cross your fingers it works for everyone. Another consideration is whether all distros have `u-boot-tools` available for installation.

On the other hand, if you want to use a consistent version of `u-boot-tools` for all distros then it should be built as part of the toolchain. You can also bump the package to a new version any time you like, to fix bugs/add new features, rather than be stuck with whatever version is installed by the various distros.

Based on earlier commits that added `u-boot-tools`, the `u-boot-tools` package is using the same source as the existing`u-boot` package which is already being built for target, so it seems you can use the `u-boot` package after adding the host side of things (along with a suitable PKG_DEPENDS_HOST).

But I've no skin in this u-boot game - I'm just making an observation that you seem to have almost all of what you need already.

Reply:
Ubuntu 16.04 provides 2016.01 and the version kszaq has been using is prior to all of this is 2016.03,  looking at the github source there was no changes to the binary in that time.

There is no need to over complicate this with extra unnecessary building of packages.

Reply:
Please don't add any more external dependencies on stuff installed on the build host. This really should be the exception. We can expect to have gcc, make, bash installed on the host but everything else should be part of the toolchain